### PR TITLE
RD-2790 Add deployment details drilldown button in Deployments View

### PR DESCRIPTION
**Pull request by Gelio**
This PR adds a "Details" button in the Deployments View widget. The button opens a page with deployment details. The page is the same as when clicking the deployment in the old Deployments widget. It contains all the widgets that are present in the details pane of the Deployments View.

Besides, I have split the existing code for the drilldown buttons to a few files.

~~Moreover, I have slightly updated the `tsconfig.json` for `widgets` to improve the autocomplete/auto-import experience when `React` is not imported when using tsserver (e.g. in neovim).~~ (reverted)

IMO it would be best to review this PR commit by commit to avoid combining a large file extraction with functional changes.

## Video

https://user-images.githubusercontent.com/889383/127465305-f27eefc1-059e-41fd-88d3-b4db03b55e2c.mp4

Note that the `LoadingOverlay` is only shown over the subdeployment drilldown buttons and the _Details_ button can be always clicked. This is because data from the backend is only required for the subdeployment drilldown buttons. Here's a video showing that when using a network slowdown (_Slow 3G_):

https://user-images.githubusercontent.com/889383/127465434-d89005e4-ed65-4aee-9146-815e9fa7643b.mp4

There is a small bug in the video (the `LoadingOverlay` is slightly offset from the drilldown buttons), but it is fixed in https://github.com/cloudify-cosmo/cloudify-stage/pull/1528/commits/e23f005aff78f9dade49ed9afeb70190ed3caaac

image

## System tests

https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1009/pipeline

Queued 2021-07-29 11:56 CEST

The only failure is the known RD-2898

### CODE REVIEW

**Comment by Gelio** [comment]
Whoops, looks like I need to fix something related to the `tsconfig.json` I modified. Regardless, the PR should be reviewable :slightly_smiling_face:

File: widgets/common/src/deploymentsView/detailsPane/drilldownButtons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+const ButtonsContainer = styled.div`
+    margin-right: 1rem;
+    margin-bottom: 1rem;
+`;
+
+const SubdeploymentButtonsContainer = styled.div`
+    position: relative;
+    // NOTE: make sure the 2 buttons appear as a single block. This way the LoadingOverlay will be a rectangle
+    // and not wrap over 2 lines.
+    display: inline-block;
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/detailsPane/drilldownButtons/index.tsx:28]: image

File: test/cypress/integration/widgets/deployments_view_spec.ts [hunk 2]
```diff
@@ -806,6 +806,21 @@ describe('Deployments View widget', () => {
             });
         });
 
+        it('should allow drilling down to deployment details', () => {
+            useEnvironmentsWidget();
+
+            getDeploymentsViewTable().contains('app-env').click();
+            getDeploymentsViewDetailsPane().contains('button', 'Details').click();
+
+            cy.get('.widget.deploymentsViewWidget').should('not.exist');
```
Comment (Gelio) [inline test/cypress/integration/widgets/deployments_view_spec.ts:815]: Ideally I would do `getDeploymentsViewWidget().should('not.exist')`, but that assertion failed due to `getDeploymentsViewWidget` using a `filter` which does not work well with `should('not.exist')`

File: widgets/common/src/deploymentsView/detailsPane/drilldownButtons/SubdeploymentDrilldownButton.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
+import type { FunctionComponent } from 'react';
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/detailsPane/drilldownButtons/SubdeploymentDrilldownButton.tsx:1]: Most of the code in this file comes from the removed `DrilldownButtons.tsx`.

AFAIR I only used `tDrillDownButtons`

File: widgets/common/src/deploymentsView/detailsPane/drilldownButtons/index.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+`;
+
+const getDeploymentUrl = (id: string) => `/deployments/${id}?all_sub_deployments=false`;
+
+const DrilldownButtons: FunctionComponent<DrilldownButtonsProps> = ({
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/detailsPane/drilldownButtons/index.tsx:33]: Most of the code here comes from the removed `DrilldownButtons.tsx`

File: widgets/common/src/deploymentsView/detailsPane/drilldownButtons/subdeployments-result.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import type { QueryObserverLoadingResult, QueryObserverSuccessResult } from 'react-query';
+import type { Deployment, DeploymentStatus } from '../../types';
+
+export type SubdeploymentsResult =
```
Comment (Gelio) [inline widgets/common/src/deploymentsView/detailsPane/drilldownButtons/subdeployments-result.ts:4]: This code comes from the removed `DrilldownButtons.tsx`

**Review by qooban — APPROVED** [review]
Nice feature 👍🏻 The change seems good to me. I have nothing to comment about.

**Comment by Gelio** [comment]
@stepek Do you plan on reviewing this PR? If you review it before the weekly meeting, I will have a chance to merge the PR and update #1529 (since there will be merge conflicts) :slightly_smiling_face: 

If not, I will leave these 2 PRs for the UI team to finalize

**Review by stepek — APPROVED** [review]

**Comment by Gelio** [comment]
Having both approvals and the system tests passing (aside from RD-2898), I'll merge this PR :slightly_smiling_face: 

Thanks for reviews :smile:

**Pull request merged**